src/metadata.rs

```diff
@@ -185,6 +185,7 @@ impl Metadata {
     /// Returns the metadata entry at the `index`.
     ///
     /// `None` is returned if out of bound.
+    #[allow(clippy::unnecessary_cast)]
```

Comment:
Why allow this?

Reply:
I added that because of this error:
error: casting raw pointers to the same type and constness is unnecessary (`*const u8` -> `*const u8`)
   --> src/metadata.rs:197:51
    |
197 |             let val_bytes = slice::from_raw_parts(val as *const u8, val_len);
    |                                                   ^^^^^^^^^^^^^^^^ help: try: `val`
on  Linux-Format job:
cargo clippy --all -- -D clippy::all && cargo clippy --all --no-default-features --features prost-codec -- -D clippy::all
If I would change that to:
    let val_bytes = slice::from_raw_parts(val, val_len);
the Win job will fail, because the warnings are treated like errors.
So either we keep that or we add conditional compilation:
#[cfg(target_os = "linux")]
 let val_bytes = slice::from_raw_parts(val, val_len);
#[cfg(target_os = "windows")]
 let val_bytes = slice::from_raw_parts(val, val_len);

something like that.

Reply:
I see. You can get around this warnings by changing it to `val.cast()`.